### DISCUSSION TITLE
Support for podman releases and EOL

### DISCUSSION THREAD

**Discussion by afbjorklund** [containers/podman#23352] [General]
I created a PR for the endoflife.date project:

* https://github.com/endoflife-date/endoflife.date/pull/5505

Based on a user request for a podman page:

* https://github.com/endoflife-date/endoflife.date/issues/3057

---

The Docker page looks like:

https://endoflife.date/docker-engine

The Podman page looks like:

https://deploy-preview-5505--endoflife-date.netlify.app/podman

----

Ideally, the endoflife.date project would llike podman.io to supply both the release timeline and the support policy:

https://endoflife.date/recommendations

Currently the dates are just taken from github releases, ~~and the supported cycles are taken from the Fedora packages:~~

https://github.com/containers/podman/releases
~~https://packages.fedoraproject.org/pkgs/podman/podman/~~

**Comment by Luap99** [comment]
I am totally in favor of creating a page where this is listed on documented.
As you know we cannot really give dates as we only support the latest release and the releases are not at a fixed interval.

In general the question is what does "supported" mean, just security updates? Using the fedora baseline is likely the most realistic for that.

But as an upstream maintainer I still ask anyone who files issues here to try to reproduce on the latest version even if there are older version in fedora and we will not create new 4.9 release for "normal" bug fixes unless someone can convince us that it is really critical.


And IMO I would really just call it Podman not Podman Engine.

**Reply by afbjorklund** [reply]
The packages for fedora and for rhel are covered on separate pages, so they were "out of scope" for the podman page:

* https://endoflife.date/fedora
* https://endoflife.date/rhel

Same for the podman packages for ubuntu and debian, that also see the (occasional) security backport of important fixes:

* https://endoflife.date/ubuntu
* https://endoflife.date/debian

**Reply by afbjorklund** [reply]
> In general the question is what does "supported" mean, just security updates?

It can be defined (and there can be several), current column says "Security Support"

**Reply by afbjorklund** [reply]
Should also mention that the packages for different distributions is already listed here:

https://repology.org/project/podman

**Reply by Luap99** [reply]
I mean in terms of security issues we only fix the latest version purely upstream speaking.

Yes there is the special case with fedora 39 and podman 4.9 right now but that is not a general rule I would say. So if the goal is to have some form of automation over there then it should really on list the latest IMO.

Of course if we create a document then we could list such cases properly and you could pull from that which I think would be the easiest thing.

**Reply by afbjorklund** [reply]
As far as I know, the automation just bumps the latest patch version within each "release" :

(https://github.com/endoflife-date/endoflife.date/commits/master/products/docker-engine.md)

```yaml
    latest: "5.1.2"
    latestReleaseDate: 2024-07-10
```

```yaml
    latest: "5.0.3"
    latestReleaseDate: 2024-05-10
```

So new cycles are still added manually, with a release date - currently matching the .0 patch:

(For docker-engine.md they are using https://github.com/moby/moby/milestones for this)

```yaml
-   releaseCycle: "5.1"
    releaseDate: 2024-05-29
    eol: false
```

```yaml
-   releaseCycle: "5.0"
    releaseDate: 2024-03-19
    eol: true
```

**Reply by afbjorklund** [reply]
Now it shows **all** releases (except the latest), as EOL. There is no special tracking of packaging in Fedora releases.

**Reply by afbjorklund** [reply]
Renamed from podman-engine to podman, and added the 5.2 and the 3.x release cycles to the version matrix.

**Comment by afbjorklund** [comment]
The timeline I created gives an indication of the average lifetime of a minor release:

https://github.com/endoflife-date/endoflife.date/issues/3057#issuecomment-2241631590

As you say, it varies a little between the releases - but on average it is rather similar:

![podman-timeline](https://github.com/user-attachments/assets/a20c45fd-40cb-4056-a27c-20ccbadbc884)

I filed a bug about the missing 5.0.0 [release blog](https://podman.io/release/).

* https://github.com/containers/podman.io/issues/316

The GitHub release notes vary a lot, in scope and in length.

Main EOL question is about 3.4 and 4.9, what date they stop?

**Reply by Luap99** [reply]
blog posts moved to https://blog.podman.io/

https://blog.podman.io/2024/03/podman-5-0-has-been-released/

**Reply by afbjorklund** [reply]
Unfortunately that breaks the links for the "pattern", like the other URL issues with the new podman.io website

Anyway, there were some notes about it (changelog and image) in the https://endoflife.date/recommendations

**Reply by afbjorklund** [reply]
Seems like the main issue was missing redirects, and missing tagging of posts with the "releases" category.

https://blog.podman.io/category/releases/
